File: InvoiceApi/ViewModels/OrderViewModel.cs
namespace InvoiceApi.ViewModels
{
    public class OrderViewModel
    {
        public Guid Id { get; set; }

        public string OrderNumber { get; set; }

        public DateTime OrderDate { get; set; }

        public CustomerViewModel Customer { get; set; }

        public List<PositionViewModel> Positions { get; set; }

        public AddressViewModel DeliveryAddress { get; set; }

        public AddressViewModel BillingAddress { get; set; }

        public double TotalPrice { get; set; }

        public bool InvoiceAddable { get; set; }
    }
}

File: invoice-fe/src/models/OrderViewModel.ts
import { AddressViewModel } from "./AddressViewModel";
import { CustomerViewModel } from "./CustomerViewModel";
import { PositionViewModel } from "./PositionViewModel";

export interface OrderViewModel {
    id: string;
    orderNumber: string;
    orderDate: Date;
    customer: CustomerViewModel;
    positions: PositionViewModel[];
    deliveryAddress: AddressViewModel;
    billingAddress: AddressViewModel;
    totalPrice: number;
    invoiceAddable: boolean;
  }